README.md

```diff
@@ -0,0 +1,131 @@
+# Maven Android
+This is a Maven Artifact Repository for uploading library `aar` files from private github repositories.
+
+## Prepare your project for deploying
+
+### General pointers before deploying your library : 
+ - Proguard your source-code preserving only the api you want to expose.
+
+### Signing your artifacts :
+In order to prove that the artifacts are uploaded by you, you need to sign them with your gpg keys. You can obtain the keys by following the steps below :
+
+1. Download GPG Suite from [here](https://releases.gpgtools.org/GPG_Suite-2016.10_v2.dmg) to create GPG keys via easy UI. Find the latest version [here](https://gpgtools.org/).
+2. Install GPG Suite.
+3. Open GPG Keychain.
+4. Click on the new Key Icon.
+5. Enter your Full Name, email, and passphrase. Check upload key checkbox. Do not lose the **passphrase**, you will need it to sign your artifacts.
+6. Click on Generate Key.
+7. After your key is generated, you will be able to see it in the keychain :
+8. Copy the **Key Id** from here. We will need it later.
+
+### Signing Information in `gradle.properties`
+Create a `gradle.properties` file inside the library module of your project (Multiple files if you have multiple modules). Set the following information obtained in the previous step :
+
+```
+signing.keyId=<yourKeyId>
+signing.password=<yourKeyPassphrase>
+signing.secretKeyRingFile=<yourKeyRingFilePath>
+```
+You can find your key file at `/Users/garima-fueled/.gnupg/secring.gpg`. Make sure to add your `gradle.properties` file to `.gitignore`.
+
+### Scripts
+ - Add the `gradle-mvn-push.gradle` and `version.gradle` gradle scripts to your project from [here](https://github.com/Fueled/garage-android/tree/master/extra/gradle). (Add it to the `extra/gradle/` directory. 
+ - Also add the deploy script `deploy_library.sh` from [here] (https://github.com/Fueled/garage-android/tree/master/scripts/). (Add it to the `scripts/` directory.
```

Comment:
@julien-fueled The link to `deploy_library.sh`  is added here, when we add the other scripts.